### ISSUE TITLE
Officially Support of PHP 8.2??

### ISSUE THREAD

**Issue by codyng** [wp-graphql/wp-graphql#2729]
### What problem does this address?

Thank you for this great plugin!! We can use GraphQL with Wordpress finally!

However, recently Azure doesn't support PHP 7.4 anymore and we are forced to upgrade to PHP 8.2.

wp-graphql generate plenty of depreciated warnings, or warnings, including "Cannot modify header information". AFAIK, it uses some coding or syntax that PHP 8.x does not support anymore.

As most of developers' experience, I think, they enable php error in debug mode and look after all these warnings.

To meet the trend, may I know wp-graphql is officially supporting PHP 8.2??

### What is your proposed solution?

Upgrade all appropriated implementations to support PHP 8.2.

**Comment by jasonbahl** [comment]
@codyng we haven't done robust testing with php 8.2 yet. (it looks like WordPress has just started publishing docker images within the last few days with 8.2, which we use in our test suites)

We do plan to work well with PHP 8.2, but we might not optimize specifically for it yet, because we also support 7.1+ (php 7.4 is most popular with our users, currently) so we need to make sure to play nice with both 7.1+ and 8+ worlds.

Can you get more specific about the warnings you're seeing and what steps we can take to reproduce? 

I'm not sure these warnings are necessarily related to PHP 8.2 or not. Would need more information / steps to reproduce in order to help determine if they are indeed related to PHP versions or not.

**Comment by codyng** [comment]
Thanks for reply. Most of the issues / warnings are related to "dynamic properties" and "incompatible return types".

What I need to do currently is to bypass the warnings by adding these attributes:
#[\AllowDynamicProperties]
#[\ReturnTypeWillChange]

Just play around after PHP 8.2 upgrade, so far what I found that the below source are affected:

wp-graphql/src/AppContext.php
wp-graphql/src/Router.php
wp-graphql/src/Type/WPConnectionType.php
wp-graphql/vendor/webonyx/graphql-php/src/Type/Definition/Type.php
wp-graphql/vendor/webonyx/graphql-php/src/Type/Definition/InputObjectField.php
wp-graphql/vendor/webonyx/graphql-php/src/Validator/Rules/ValuesOfCorrectType.php
wp-graphql/vendor/webonyx/graphql-php/src/Language/SourceLocation.php
wp-graphql/vendor/webonyx/graphql-php/src/Language/AST/NodeList.php
wp-graphql/vendor/webonyx/graphql-php/src/Error/Error.php
wp-graphql/vendor/webonyx/graphql-php/src/Executor/ExecutionResult.php

**Comment by elliottpost** [comment]
On a related note I am getting _deprecation warnings_ (not errors, yet, but I think PHP 8.2 bumps then into error territory) with PHP 8.1 that I think upgrading to a newer version of WebOnyx would solve. 

Warnings:
```
Deprecated: Return type of GraphQL\Type\Definition\Type::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Type/Definition/Type.php on line 311 Deprecated: Return type of GraphQL\Language\SourceLocation::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Language/SourceLocation.php on line 49 Deprecated: Return type of GraphQL\Language\AST\NodeList::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Language/AST/NodeList.php on line 72 Deprecated: Return type of GraphQL\Executor\ExecutionResult::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /app/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ExecutionResult.php on line 118
```

It looks like WebOnyx 14.11.4 would _potentially_ solve the issue as they changed their PHP support flag from PHP ^8 to PHP ^8.0 (https://packagist.org/packages/webonyx/graphql-php#v14.11.4) and still support PHP 7.1+, but I haven't tested that yet. I suspect for PHP 8.2+ WebOnyx 15+ would be needed which requires PHP 7.4+

**Comment by jasonbahl** [comment]
I believe we can consider this resolved after #2818. 

If there's any specific errors re: WPGraphQL and PHP 8.2 support, please open a new issue with steps to reproduce. 🙏🏻

**Comment by justlevine** [comment]
(note to self: confirm Models don't throw a deprecation notice in 8.2+ )
